ion-schema/src/types.rs

```diff
@@ -241,79 +241,102 @@ impl Display for BuiltInTypeDefinition {
     }
 }
 
-/// Represents a [`TypeDefinition`] which stores a resolved ISL type using [`TypeStore`]
+/// Represents a [`TypeDefinitionKind`] which stores a resolved ISL type using [`TypeStore`]
 #[derive(Debug, Clone, PartialEq)]
-pub enum TypeDefinition {
+pub enum TypeDefinitionKind {
     Named(TypeDefinitionImpl),
     Anonymous(TypeDefinitionImpl),
     BuiltIn(BuiltInTypeDefinition),
 }
```

Comment:
Yes,  I was thinking to do something similar because the `TypeDefinitionImpl` here is repetitive and uses `Option<String>` for naming. I will add an issue for this into the milestone.

Reply:
https://github.com/amazon-ion/ion-schema-rust/issues/166